src/frontend/src/catalog/system_catalog/mod.rs

```diff
@@ -116,6 +121,89 @@ impl SysCatalogReaderImpl {
     }
 }
 
+pub struct BuiltinTable {
+    name: &'static str,
+    schema: &'static str,
+    columns: &'static [SystemCatalogColumnsDef<'static>],
+    pk: &'static [usize],
+}
+
+pub struct BuiltinView {
+    name: &'static str,
+    schema: &'static str,
+    columns: &'static [SystemCatalogColumnsDef<'static>],
+    sql: String,
+}
+
+pub enum BuiltinCatalog {
+    Table(&'static BuiltinTable),
+    View(&'static BuiltinView),
+}
+
+impl BuiltinCatalog {
+    fn name(&self) -> &'static str {
+        match self {
+            BuiltinCatalog::Table(t) => t.name,
+            BuiltinCatalog::View(v) => v.name,
+        }
+    }
+}
+
+impl From<&BuiltinTable> for SystemTableCatalog {
```

Comment:
May I ask what the difference will be between `BuiltinTable` and `SystemTableCatalog`? 👀

Reply:
`BuiltinTable` is used by kernel developers to define a builtin system table with some necessary fields in a more concise way. `SystemTableCatalog` is the real catalog which is used by the frontend with some extra infos like `id`, `ColumnCatalog`s and `owner`. These infos will be filled when initialized.